### ISSUE TITLE
Sendgrid D7 API composer.lock error handling

### ISSUE THREAD

**Issue by rachelwhitton** [pantheon-systems/documentation#4611]
Re: [Using SendGrid To Deliver Email](https://pantheon.io/docs/guides/sendgrid/)

Priority: Medium 

## Issue Description:
After successfully configuring API integration on a D7 site, email tests are successful - however a composer.lock error is shown across all environments 

## Suggested Resolution
Add troubleshooting section that provides context on how to handle composer.lock error 
<img width="1149" alt="Screen_Shot_2019-03-20_at_3_00_36_PM" src="https://user-images.githubusercontent.com/10119525/54952869-faf7e100-4f14-11e9-96e9-bc07aaaafd74.png">

**Comment by rachelwhitton** [comment]
@stevector have you observed this before? Any tips?

**Comment by enotick** [comment]
Just tested it out and I don't get the error. I'd check to see if composer.lock file was generated after you ran `composer require fastglass/sendgrid:~1.0` if it didn't and you see the following section in your "require": {
    "fastglass/sendgrid": "~1.0"
  }
try `composer install` or `composer update` ; if it's not in the composer.json run the require command again; Then also check when you are pushing to git that composer.lock is committed and not excluded in your .gitignore

**Comment by newcircle** [comment]
I still see this error in all three of our environments: dev, stage and prod site.

Not sure if this has anything to do with how this was set up --  Rachel helped us configure Sendgrid via a Multi-dev branch first, and later one we merged to dev, then pushed to test and prod.

**Comment by rachelwhitton** [comment]
Thanks @newcircle and @enotick! I appreciate the discussion 

Note in this case `composer.lock` exists in all three environments at the docroot, alongside `composer.json`

**Comment by alexfornuto** [comment]
I've also tested this issue and found not problems. Assigning to @rachelwhitton who's gonna review my test environment.

**Comment by enotick** [comment]
And if you look at the serve do you have the .lock file in the directory?
And if you do maybe it is a permissions problem, I'd probably just look
into permissions for the autoload file and change .lock to match. Hope it
helps :)

чт, 25 апр. 2019 г. в 12:25, Rachel <notifications@github.com>:

**Comment by alexfornuto** [comment]
Upon closer review, I noticed that the error specified that it was looking for `composer.lock` in `sites/all`, but the actual file is in the code root. To solve, I symlinked the file:

```bash
cd sites/all
ln -s ../../composer.lock ./composer.lock
```

This "works" to remove the error, but it sure seems hacky. I'm not sure if a Drupal Core update would overwrite or have problems from this symlink.

Pinking @greg-1-anderson for his input.
